x-pack/plugins/maps/public/classes/sources/es_geo_grid_source/es_geo_grid_source.tsx

```diff
@@ -303,55 +354,72 @@ export class ESGeoGridSource extends AbstractESAggSource {
       }),
     });
 
-    return convertRegularRespToGeoJson(esResponse, this._descriptor.requestType);
+    return convertRegularRespToGeoJson(esResponse, this._descriptor.requestType || RENDER_AS.POINT);
   }
 
-  async getGeoJsonWithMeta(layerName, searchFilters, registerCancelCallback, isRequestStillActive) {
-    const indexPattern = await this.getIndexPattern();
-    const searchSource = await this.makeSearchSource(searchFilters, 0);
+  async getGeoJsonWithMeta(
+    layerName: string,
+    searchFilters: VectorSourceRequestMeta,
+    registerCancelCallback: (callback: () => void) => void,
+    isRequestStillActive: () => boolean
+  ): Promise<GeoJsonWithMeta> {
+    const indexPattern: IndexPattern = await this.getIndexPattern();
+    const searchSource: ISearchSource = await this.makeSearchSource(searchFilters, 0);
 
     let bucketsPerGrid = 1;
     this.getMetricFields().forEach((metricField) => {
       bucketsPerGrid += metricField.getBucketCount();
     });
 
-    const features =
-      bucketsPerGrid === 1
-        ? await this._nonCompositeAggRequest({
-            searchSource,
-            indexPattern,
-            precision: searchFilters.geogridPrecision,
-            layerName,
-            registerCancelCallback,
-            bufferedExtent: searchFilters.buffer,
-          })
-        : await this._compositeAggRequest({
-            searchSource,
-            indexPattern,
-            precision: searchFilters.geogridPrecision,
-            layerName,
-            registerCancelCallback,
-            bucketsPerGrid,
-            isRequestStillActive,
-            bufferedExtent: searchFilters.buffer,
-          });
+    let features: Feature[];
+    if (searchFilters.buffer) {
```

Comment:
Why was this if statement added? May be the method should throw if buffer is not provided?